lib/fauna/errors.rb

```diff
@@ -91,8 +72,97 @@ class InternalError < FaunaError; end
   # An exception thrown if FaunaDB responds with an HTTP 503.
   class UnavailableError < FaunaError; end
 
+  # :section: ErrorData
+
+  # Data for one error returned by the server.
+  class ErrorData
+    def self.from_hash(hash)
+      code = hash[:code].to_sym
```

Comment:
I don't know that I like symbolizing the codes (or the position either). Seems like it'll just be a pain, cause the codes basically all have spaces or what have you, and it's inconsistent to me that some are symbols and some aren't